core/creators/themedui.js

```diff
@@ -384,6 +388,63 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayDetached` to be `true`.
+	//
+	// If `config.registerCallback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayDetached: true,
+	//		delayDetachedFrequency: 50,
+	//		registerCallback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	// @param {String} mode CKEditor mode.
+	function delayCreationOnDetachedElement( element, config, data, mode ) {
+		if ( !config || !config.delayDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		if ( !config.registerCallback ) {
+			CKEDITOR.config.registerCallback( payload );
+			return true;
+		}
+
+		config.registerCallback( registerPayload );
+
+		return true;
+
+		function payload() {
+			var delay = parseFloat( config.delay );
+			delay = isNaN( delay ) ? CKEDITOR.config.delayDetachedFrequency : delay;
+
+			var intervalId = setInterval( function() {
```

Comment:
As we discussed in team talk, let's add a `CKEDITOR.warn()` message before the first interval and when the interval is removed (so editor is created). Please remember to test it (probably stubbing the `CKEDITOR.warn()` method with Sinon).

Reply:
Probably we should replace the test message with a new error-code?

The editor displays a tip, that more information could be in docs and also warn the function itself to treat the first argument as an error code - not a message.

Reply:
Yes, I think it should be defined as a standard error and the delay set in config should be passed as additional data. IIRC you've already mentioned before that https://github.com/ckeditor/ckeditor4-docs/pull/344 should be closed due to the change of implementation, so I've created a separate issue for adding error codes for this PR (https://github.com/ckeditor/ckeditor4/issues/4564).

Please note that what you wrote about `CKEDITOR.warn()` also applies to the success info, so it should also be extracted as an error code (I've had some doubts since it's a positive feedback for user, but we also use those codes for the warnings and this can be treated as a warning, so I think we are fine here).